services/tables/src/main/java/com/linkedin/openhouse/tables/repository/impl/InternalRepositoryUtils.java

```diff
@@ -35,14 +34,19 @@ private InternalRepositoryUtils() {
     // Utils' class constructor, noop
   }
 
-  public static java.nio.file.Path constructTablePath(
+  public static URI constructTablePath(
       StorageManager storageManager, String databaseID, String tableId, String tableUUID) {
     // TODO: Default storage is used here. Support for non-default storage type per table needs to
     // be added.
-    return Paths.get(
-        storageManager.getDefaultStorage().getClient().getRootPrefix(),
-        databaseID,
-        tableId + "-" + tableUUID);
+    String endpoint = storageManager.getDefaultStorage().getClient().getEndpoint();
```

Comment:
There's no need to make this change yet. constructTablePath is only getting used in CTAS (and only for hdfs). non-hdfs storages need much more things to correctly support ctas: BDP-23723